test/unit/alignment/pairwise/pairwise_alignment_collection_callback_test_template.hpp

```diff
@@ -0,0 +1,111 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#pragma once
+
+#include <gtest/gtest.h>
+
+#include <seqan3/std/algorithm>
+#include <seqan3/std/ranges>
+
+#include <seqan3/alignment/pairwise/align_pairwise.hpp>
+#include <seqan3/range/views/to_char.hpp>
+#include <seqan3/range/views/zip.hpp>
+#include <seqan3/test/expect_range_eq.hpp>
+
+#include "fixture/alignment_fixture.hpp"
+
+using namespace seqan3::test::alignment::fixture;
+
+template <auto _fixture>
+struct pairwise_alignment_fixture : public ::testing::Test
+{
+    auto fixture() -> decltype(alignment_fixture_collection{*_fixture}) const &
+    {
+        return *_fixture;
+    }
+};
+
+template <typename fixture_t>
+class pairwise_alignment_collection_callback_test : public fixture_t
+{};
+
+TYPED_TEST_SUITE_P(pairwise_alignment_collection_callback_test);
+
+TYPED_TEST_P(pairwise_alignment_collection_callback_test, score)
+{
+    auto const & fixture = this->fixture();
+    seqan3::configuration align_cfg = fixture.config |
+                                      seqan3::align_cfg::result{seqan3::with_score} |
+                                      seqan3::align_cfg::on_result{[&] (auto && result)
```

Comment:
Is there a way to get a the type of `result` beforehand?

Reply:
Not yet, and I thought about it and I couldn't find an elegant solution quicly. It is kind of a henn-and-egg problem. I need the configuration type to determine the result type. But I can't have the configuration type only without the on_result config where I might need the type. So this needs some more thoughts on the proper design.